### PR TITLE
Oas 4894 | Add Data Source for supported Cloud Providers 

### PR DESCRIPTION
**Pull request by adoi**

### CODE REVIEW

File: examples/cloud_provider/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Example: Oasis Cloud Providers
+
+This example shows how to use the Terraform Oasis provider to get the list of support cloud providers in Oasis.
+
+## Prerequsites
```
Comment (ewoutp) [inline examples/cloud_provider/README.md:5]: -> Prerequisites

File: examples/cloud_provider/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+## Environment variables
+Please refer to [Main README](../../README.md) file for all the environment variables you might need.
+
+## Instructions on how to run:
+```
+terraform init
+terraform plan
+terraform apply
+```
```
Comment (ewoutp) [inline examples/cloud_provider/README.md:18]: IMO it would help so show example output

**Review by forgems — APPROVED** [review]
LGTM

File: pkg/data_source_cloud_providers.go [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			providerProvidersFieldName: {
+				Type:        schema.TypeList,
+				Description: "List of all supported cloud providers in Oasis.",
+				Computed:    true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
```
Comment (ewoutp) [inline pkg/data_source_cloud_providers.go:57]: I'm missing provider ID

File: examples/cloud_provider/output.tf [hunk 4]
```diff
@@ -0,0 +1,9 @@
+// Load in Oasis terms and conditions
```
Comment (forgems) [inline examples/cloud_provider/output.tf:1]: Is this comment correct ?

**Review by ewoutp — CHANGES_REQUESTED** [review]

**Review by ewoutp — APPROVED** [review]
LGTM

**Pull request merged**